llvm/lib/Analysis/ConstraintSystem.cpp

```diff
@@ -95,17 +95,19 @@ bool ConstraintSystem::eliminateUsingFM() {
           IdxUpper++;
         }
 
-        if (MulOverflow(UpperV, ((-1) * LowerLast), M1))
```

Comment:
Another question: Does `(-1) * LowerLast` never overflow?

Reply:
Ping.

Reply:
I think it might be possible to construct a case where it overflows. Ideally we would construct such a case, add a test and check for overflow.